### PR TITLE
fix(eds): WriteEDS thread safety for concurrent writingSessions

### PR DESCRIPTION
**Pull request by distractedm1nd**
## Overview
This PR removes usage of nmt's defaultHasher to allow for concurrent writingSessions.

## Checklist

- [X] New and updated code has appropriate documentation
- ~~New and updated code has new and/or updated testing~~
- [x] Required CI checks are passing
- ~~Visual proof for any user facing features like CLI or documentation updates~~
- ~~Linked issues closed with keywords~~

### CODE REVIEW

File: share/eds/eds.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package eds
 import (
 	"bytes"
 	"context"
+	"crypto/sha256"
```
Comment (Wondertan) [inline share/eds/eds.go:6]: pls use minio dep instead

Comment (Wondertan) [reply share/eds/eds.go:6]: check it out namespace hasher

**Review by vgonkivs — APPROVED** [review]

**Review by Wondertan — CHANGES_REQUESTED** [review]
LGTM, just one thing needs to be changed

File: share/eds/eds.go (hunk 1 above)
Comment (distractedm1nd) [reply share/eds/eds.go:6]: https://github.com/celestiaorg/nmt/blob/95e8313eb788b46b30f3f85b20fd9a40bf68d432/hasher.go#L5

Comment (Wondertan) [reply share/eds/eds.go:6]: I know, but we use a different, more efficient one. The NMT lib does not import it directly not to bloat go.mod, while it still allows setting custom hasher exactly for this case.

Comment (distractedm1nd) [reply share/eds/eds.go]: done

**Review by Wondertan — APPROVED** [review]
🚢 

One last thing is to double check other usages of `nmt.Sha256Namespace8FlaggedLeaf`

**Comment by distractedm1nd** [comment]
yup, checked, no other usages

File: share/eds/eds.go [hunk 2]
```diff
@@ -28,16 +29,19 @@ import (
 	"github.com/celestiaorg/celestia-node/share/ipld"
 )
 
+// childHashLen is the length of each half of an inner node's data. It consists of the respective
+// min and max namespaces, followed by a sha256 hash.
+const childHashLen = (appconsts.NamespaceSize * 2) + sha256.Size
```
Comment (walldiss) [inline share/eds/eds.go:34]: I wonder if we should exposes `ipld.nmtHashSize` and use it here, it is the same.

Comment (walldiss) [reply share/eds/eds.go:34]: + for exposing is that I use it in `shrex/nd`. So that would be 2 usages already

Comment (distractedm1nd) [reply share/eds/eds.go]: good find, done

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]
LGTM

**Review by renaynay — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**